doc/environment.yml

```diff
@@ -0,0 +1,18 @@
+dependencies:
```

Comment:
I don't think we should maintain these `enviornment.yml` files, the information here is basically the combination of what's in `setup.py` and `pyproject.toml`, and I just don't think anyone will be updating these over time since none of us use Windows or conda.

I noticed that you needed `libpgmath` for Windows, that sort of information may be better suited in the docs.

Reply:
Okay. I'm going to try beefing up the requirements section in the docs then if that's alright. When looking to build the package from source my first time, it was a little confusing having requirements spread across the install instructions. An `environment.yml` is really nice for my sanity but I understand the maintenance burden

Reply:
Yeah I know, it's just that we don't really use conda and I hesitate to officially support it, especially the more complicated stuff with cross-compilation. I was hoping that between the docs and the conda-forge package that would cover most people's needs.

Reply:
That makes total sense. Yeah I'll remove it and add some docs for the requirements!

Reply:
Do we want to leave in one environment.yml for Windows CI?

Reply:
Yeah if it's needed for the CI then we can leave it in. However, I would prefer if it goes under the `.github` directory instead of its current location.

Reply:
Will do. I'll move it when I add the CI stuff for Windows

Reply:
@nwu63 So I'm not sure how to get Windows CI working. Below is the Azure job I was going to add, but idk where to add it

```yaml
- job: Windows
    pool:
      vmImage: windows-latest

    - powershell: Write-Host "##vso[task.prependpath]$env:CONDA\Scripts"
      displayName: Add conda to PATH

    - script: conda config --add channels conda-forge && conda config --set channel_priority strict
      displayName: Set channel priority

    - script: conda create --yes --name pyos-build
      displayName: Create environment

    - script: |
        call activate pyos-build
        call conda install -y mamba
        call mamba env update --file .github/environment.yml
        call mamba install -y libpgmath
      displayName: Install mamba and update environment

    - script: |
        set IPOPT_DIR=%CONDA_PREFIX%\Library
        set CC=cl
        set FC=flang
        set CC_LD=link
        python -m build -n -x .
        pip install --no-deps --no-index --find-links dist pyoptsparse
      displayName: Build and install pyoptsparse

    - script: |
        cd tests
        testflo -n 1 .
      displayName: Run tests
```

Reply:
Hmm so this is a bit tricky, because as you see we use templates for our Pipelines. I'm not sure what's the easiest way to add a separate, standalone job to the existing stuff, do we need a separate YAML file and a separate Pipeline? Perhaps it's easiest if we use GitHub actions instead?

Reply:
I tried adding a GitHub Action and it says it needs to be approved. Hoping that works! Also, I accidentally merged main into meson instead of rebasing the meson branch onto main. Let me know if you want to me to clean the history up

Reply:
I just approved the workflow. Also, we squash the commits, so merging is fine.

Reply:
Sounds good. Looks like any change I make requires a new approval...

Reply:
I've temporarily adjusted permissions so that it should not require new approval. Also, maybe we can use [this action](https://github.com/marketplace/actions/setup-miniconda) to set up miniconda in GHA?

Reply:
Ah yes thank you. I found that a few days ago but couldn't find it again today. I will make that change

Reply:
Windows tests pass! What from here?

Reply:
Nice! Two minor things for the GHA:
1. I think it'd be nice to have `--pre_announce -v` added to the testflo command for Windows so that we can see which tests were skipped
2. Is it possible to install IPOPT via conda on Windows? That way we can test the IPOPT wrapper also

Reply:
Great! Once this PR is into main, I will update the conda-forge feedstock to point to main and to have the correct version. Regarding your comments:

1. I can add this when I'm back at my desk. 
2. IPOPT should already be getting built and tested. It's installed through conda and the proper environment variable is set. I admit I didn't check to make sure the tests for IPOPT were getting run. Are you sure they're not getting run?

Reply:
Sorry, I realized for IPOPT you had no way of knowing cause the tests weren't shown. The requested flags were added and IPOPT is indeed in there

Reply:
Ah right I forgot that IPOPT is in the conda YAML file already. Exactly, now we can check and make sure they're getting tested. Unfortunately in the current implementation if the imports fail we just skip those optimizers, which may mask actual installation issues.